code/game/objects/structures/safe.dm

```diff
@@ -2,80 +2,110 @@
 CONTAINS:
 SAFES
 FLOOR SAFES
-Safe Codes
-Safe Internals
+SAFE INTERNALS
+SAFE CODES
 */
 
 GLOBAL_LIST_EMPTY(safes)
 
 //SAFES
 /obj/structure/safe
 	name = "\improper Safe"
-	desc = "A huge chunk of metal with a dial embedded in it. Fine print on the dial reads \"Scarborough Arms - 2 tumbler safe, guaranteed thermite resistant, explosion resistant, and assistant resistant.\""
+	desc = "A huge chunk of metal with a dial embedded in it. Fine print on the dial reads \"Scarborough Arms tumbler safe, guaranteed thermite resistant, explosion resistant, and assistant resistant.\""
 	icon = 'icons/obj/structures.dmi'
 	icon_state = "safe"
-	anchored = 1
-	density = 1
-	var/open = FALSE	//is the safe open?
+	
+	anchored = TRUE
+	density = TRUE
+	resistance_flags = LAVA_PROOF | FIRE_PROOF
+	unacidable = TRUE
+
+	var/open = FALSE
 	var/locked = TRUE
-	var/tumbler_1_pos	//the tumbler position- from 0 to 72
-	var/tumbler_1_open	//the tumbler position to open at- 0 to 72
-	var/tumbler_2_pos
-	var/tumbler_2_open
-	var/open_pos
-	var/dial = 0		//where is the dial pointing?
-	var/space = 0		//the combined w_class of everything in the safe
-	var/maxspace = 24	//the maximum combined w_class of stuff in the safe
+	var/dial = 0		// The position the dial is pointing to.
+
+	var/number_of_tumblers = 3 // The amount of tumblers that will be generated.
+	var/list/tumblers = list() // The list of tumbler dial positions that need to be hit.
+	var/list/current_tumbler_index = 1 // The index in the tumblers list of the tumbler dial position that needs to be hit.
+
+	var/space = 0		// The combined w_class of everything in the safe.
+	var/maxspace = 24	// The maximum combined w_class of stuff in the safe.
+
 	var/obj/item/thermal_drill/drill = null
 	var/drill_timer
 	var/time_to_drill
 	var/image/bar
 	var/drill_start_time
 	var/drill_x_offset = -13
 	var/drill_y_offset = -3
-	var/knownby = list()
-
+	var/known_by = list()
 
 /obj/structure/safe/New()
 	GLOB.safes += src
-	tumbler_1_pos = rand(0, 71)
-	tumbler_1_open = rand(0, 71)
-	
-	tumbler_2_pos = rand(0, 71) // first value in the combination set first
-	tumbler_2_open = rand(0, 71)
 
-/obj/structure/safe/Initialize()
-	..()
+	for(var/i = 1 to number_of_tumblers)
+		tumblers.Add(rand(0, 99))
+
+/obj/structure/safe/Initialize(mapload)
+	. = ..()
+
 	for(var/obj/item/I in loc)
 		if(space >= maxspace)
 			return
 		if(I.w_class + space <= maxspace)
 			space += I.w_class
-			I.loc = src
+			I.forceMove(src)
+
+/obj/structure/safe/Destroy()
+	GLOB.safes -= src
+	drill?.soundloop?.stop()

```

Comment:
Funky.

Reply:
I wasn’t actually aware that BYOND added this null operator before this! That operator was already present in the code. Fancy indeed though!

Reply:
As the person who originally used added operator here I also approve 👍

Reply:
That looks so freaky, never seen that before